### ISSUE TITLE
Issue with secrets_masker.py

### ISSUE THREAD

**Issue by jerell-mendoza** [apache/airflow#32617]
### Apache Airflow version

2.6.3

### What happened

While running an ETL we are getting this warning in the logs

`[2023-07-14, 19:12:32 UTC] {trigger_dagrun.py:197} INFO - Waiting for v2_mdcr_por_org_bene_dtl on 2023-07-14T19:11:31.849123+00:00 to become allowed state [<TaskInstanceState.SUCCESS: 'success'>] ...

[2023-07-14, 19:12:32 UTC] {secrets_masker.py:273} WARNING - Unable to redact <TaskInstanceState.SUCCESS: 'success'>, please report this via <https://github.com/apache/airflow/issues>. Error was: TypeError: EnumMeta.__call__() missing 1 required positional argument: 'value'

[2023-07-14, 19:13:32 UTC] {secrets_masker.py:273} WARNING - Unable to redact <TaskInstanceState.SUCCESS: 'success'>, please report this via <https://github.com/apache/airflow/issues>. Error was: TypeError: EnumMeta.__call__() missing 1 required positional argument: 'value'`

We are using the `TriggerDagRunOperator` and this Task is waiting for another Dag to finish and we see this log repeatedly showing until the Dag is finished. We are using the param `wait_for_completion=True` on this operator. As a note, this `TriggerDagRunOperator` is triggering another Dag that contains several other `TriggerDagRunOperator`. I.e. 1 Dag, Triggers several Dags, one of those Dags triggers 7 dags.

### How to reproduce

I don't think we are doing anything special, we are running our ETLs and using the `TriggerDagRunOperator` - and it is during the waiting period for the task to finish which we are seeing this issue.

### Operating System

Debian GNU/Linux 11 (bullseye)

### Versions of Apache Airflow Providers

```
apache-airflow-providers-amazon==8.2.0
apache-airflow-providers-celery==3.2.1
apache-airflow-providers-common-sql==1.5.2
apache-airflow-providers-ftp==3.4.2
apache-airflow-providers-http==4.4.2
apache-airflow-providers-imap==3.2.2
apache-airflow-providers-jdbc==4.0.0
apache-airflow-providers-pagerduty==3.3.0
apache-airflow-providers-postgres==5.5.1
apache-airflow-providers-snowflake==4.2.0
apache-airflow-providers-sqlite==3.4.2
apache-airflow-providers-ssh==3.7.1
```

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

K8s

### Anything else

This happens for every dag run that we are doing.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by hussein-awala** [comment]
I cannot reproduce the issue, could you provide an example dag which have the same issue?

**Comment by jerell-mendoza** [comment]
@hussein-awala sorry for such a late response!

```
with DAG(
    dag_id=dag_id,
    max_active_runs=1,
    start_date=datetime(2020, 8, 5),
    schedule=schedule,
    catchup=False,
    params=params,
    default_args=get_dag_defaults(log_pagerduty=False),
    tags=["orchestration"],
) as dag:
... (there are several trigger dag run operators here, but here's one)
  TriggerDagRunOperator(
          task_id="trigger_orchestration_source_etls",
          trigger_dag_id="orchestration_source_etls",
          wait_for_completion=True,
      )
  ...
 ```
 
 This is code for `orchestration_source_etls` DAG
 
 ```
 with DAG(
        dag_id=dag_id,
        schedule=schedule,
        max_active_runs=1,
        max_active_tasks=max_parallelism,
        catchup=False,
        params=params,
        default_args=get_dag_defaults(log_pagerduty=False),
        tags=tags,
    ) as dag:
        orchestrated_weighted_dags = get_weighted_dags(orchestrated_dag_ids)
        for etl_dag in orchestrated_weighted_dags:
            dag_name = etl_dag["dag_name"]
            TriggerDagRunOperator(
                task_id=f"trigger_{dag_name}",
                trigger_dag_id=dag_name,
                priority_weight=etl_dag["weight"],
                weight_rule="absolute",
                wait_for_completion=True,
            )
 ```

Sorry trying not to reveal too much of our code - but in essence it's 
 
Top Level Dag => 
Triggers Dags [a, b, c, d, etc] => 
[
Dag a => Triggers Dags [e, f, g, etc]
Dag b => Triggers Dags [i, j, k, etc]
]

If I have time maybe I can copy and generalize our code more and see if I can reproduce it with simple dags that keep triggering other dags

**Comment by jerell-mendoza** [comment]
@hussein-awala I also noticed, something interesting.

In the same ETL for Airflow 2.6.0 I see these logs
<img width="219" alt="Screenshot 2023-08-03 at 9 17 48 AM" src="https://github.com/apache/airflow/assets/12398712/5f43a74a-2f32-4d5b-b016-9bd16d3f0f1d">

However in 2.6.3, I see these logs
<img width="373" alt="Screenshot 2023-08-03 at 9 17 42 AM" src="https://github.com/apache/airflow/assets/12398712/9c654ca8-e867-4761-970a-6f6b38c1fe3c">

**Comment by hussein-awala** [comment]
we just merged a PR to fix this issue.
Fixed by #33065

**Comment by jerell-mendoza** [comment]
@hussein-awala amazing ty - glad we got this resolved. Do you happen to know when this could get released out?

**Comment by potiuk** [comment]
> @hussein-awala amazing ty - glad we got this resolved. Do you happen to know when this could get released out?

We are aiming to release 2.7.0 beta 1 in the coming days - maybe even today or tomorrow - watch for devlist announcements and request to help with testing. And the answer to your question is very simple: If you help to test it and report any issues found with betta release, it will be faster, if we don't have much help from you and users like you to test it, it will take longer to iron out any issues. The more you help and test, the faster it will be released.

**Comment by dschiavu** [comment]
Just a quick question: does this bug affect the DAG execution in any way, i.e. does it have any significant impact on the normal running of DAGs that are using `TriggerDagRunOperator`s?

I have been getting the same `{secrets_masker.py:273} WARNING - Unable to redact <TaskInstanceState.SUCCESS: 'success'>, please report this via https://github.com/apache/airflow/issues. Error was: TypeError: EnumMeta.call() missing 1 required positional argument: 'value'` message in some of my DAGs as well.

I'd prefer to avoid upgrading to 2.7.0 where this is fixed, if possible. Thanks.

**Comment by potiuk** [comment]
I'd strongly encourage you to upgrade to latest version of Airflow since you have good reason. There are many tens or hundreds of bug fixes implemented since the version you have - depends on verssion, and there are quite a few security vulnerabilities implemented in 2.7.* line so It's a smart idea to upgrade (and keep on upgrading as we release new versions for the same reasons). 

You might be affected by any of those bugs fixed (including security advisories) and definitely the best way to address them is to upgrade things. Airflow follows SemVer and any 2.* version is intended to be backwards compatible (occassionally things are changing and are mentioned in relase notes - mostly having good reasons, being security related things. 

And you even can if you want to keep any providers you use in their old versions even if you are afraid of backwards incompatible changes in providers (which might happen).

And answering your question - probably yes, at the very least it impacts performance by the sheer fact of writing the logs. There might be also other things. Not mentioning the tens of fixes you would get when upgrading.

BTW. There are regulations being just passed in EU (Cyber Resilience Act) and US that will make all companies require to upgade their software to be upgraded when there are security fixes, so in the next 24 months or so you will likely have to start doing regular updates, so I also encourage people to start doing such regular upgrades now to get used to it and work out processes that they will have ot deploy soon anyway.

**Comment by dschiavu** [comment]
Thanks for your detailed reply, @potiuk . 

Definitely agree with your assertion - it's always a good practice to upgrade to the latest versions as early as possible, where feasible.

It appears this bug is not affecting my DAGs functionally, but as you said, it could have performance implications (we are not experiencing any, so far). 

As for security issues, I am actually running AWS's managed Airflow distribution (the AWS MWAA service) so I presume AWS takes care of the security hotfixes and backports, plus I am running it in a ring fenced environment. MWAA's latest Airflow version available is 2.7.0 anyway.
